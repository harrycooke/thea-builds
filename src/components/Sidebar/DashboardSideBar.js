// import React from "react";
// import {
//   Drawer,
//   List,
//   ListItem,
//   ListItemIcon,
//   ListItemText,
// } from "@material-ui/core";
// import SettingsIcon from "@material-ui/icons/Settings";
// import PeopleIcon from '@material-ui/icons/People';
// import BarChartIcon from '@material-ui/icons/BarChart';
// import LayersIcon from '@material-ui/icons/Layers';
// import { NavLink } from "react-router-dom";
// import { withStyles } from "@material-ui/core/styles";
// import classNames from "classnames";

// const drawerWidth = 280;

// const styles = theme => ({
//   drawerPaper: {
//     position: "fixed",
//     top: 0,
//     whiteSpace: "nowrap",
//     width: drawerWidth,
//     backgroundColor: theme.palette.primary.main,
//     transition: theme.transitions.create("width", {
//       easing: theme.transitions.easing.sharp,
//       duration: theme.transitions.duration.enteringScreen
//     })
//   },
//   drawerPaperClose: {
//     overflowX: "hidden",
//     transition: theme.transitions.create("width", {
//       easing: theme.transitions.easing.sharp,
//       duration: theme.transitions.duration.leavingScreen
//     }),
//     width: theme.spacing(8),
//     [theme.breakpoints.up("sm")]: {
//       width: theme.spacing(7)
//     }
//   },
//   media: {
//     height: 0,
//     paddingTop: '13.33%', // 16:9,
//     margin: 15,
//   },
//   mediaClose: {
//     height: 0,
//     paddingTop: '13.33%', // 16:9,
//     margin: 15,
//     display: "none"
//   },
//   listBox: {
//     position: "fixed",
//     top: theme.spacing(8),
//   },
//   listItemTextRoot: {
//     padding: 0
//   },
//   listItemTextLabel: {
//     color: "#FFFFFFF"
//   }
// });

// const DashboardSideBar = props => {
//   const { open, classes, handleToggleDrawer } = props;

//   return (
//     <Drawer
//       variant="permanent"
//       classes={{
//         paper: classNames(
//           classes.drawerPaper,
//           !open && classes.drawerPaperClose
//         )
//       }}
//       open={open}
//       onMouseOver={handleToggleDrawer}
//       onMouseOut={handleToggleDrawer}
//     >
//       <div style={{
//         marginLeft: 20,
//         // display: open
//         display: open ? "flex" : "none"
//       }}>
//         <p style={{
//           color: "#23C8DE",
//           fontFamily: "Lato",
//           fontWeight: "bold",
//           fontSize: 45
//         }}>
//           Thea
//         </p>
//         <p style={{
//           color: "#2F5496",
//           fontFamily: "Lato",
//           fontWeight: "bold",
//           fontSize: 45
//         }}>
//           Health
//         </p>
//       </div>

//       <List style={
//         open ? { marginTop: "30px" }
//           : { marginTop: "70px" }
//       }>
//         <NavLink to="/">
//           <ListItem button>
//             <ListItemIcon>
//               <SettingsIcon style={{ color: "#b0bdd2" }} />
//             </ListItemIcon>
//             <ListItemText
//               primary="Summary"
//               dense={classes.listItemTextLabel}
//               classes={{
//                 root: classes.listItemTextRoot,
//                 primary: classes.listItemTextLabel
//               }} />
//           </ListItem>
//         </NavLink>
//         <NavLink to="/dashboard/utilization">
//           <ListItem button>
//             <ListItemIcon>
//               <PeopleIcon style={{ color: "#b0bdd2" }} />
//             </ListItemIcon>
//             <ListItemText
//               primary="Utilization"
//               classes={{
//                 root: classes.listItemTextRoot,
//                 primary: classes.listItemTextLabel
//               }} />
//           </ListItem>
//         </NavLink>
//         <NavLink to="/">
//           <ListItem button>
//             <ListItemIcon>
//               <BarChartIcon style={{ color: "#b0bdd2" }} />
//             </ListItemIcon>
//             <ListItemText
//               primary="Savings & Reimbursements"
//               classes={{
//                 root: classes.listItemTextRoot,
//                 primary: classes.listItemTextLabel
//               }} />
//           </ListItem>
//         </NavLink>
//         <NavLink to="/">
//           <ListItem button>
//             <ListItemIcon>
//               <LayersIcon style={{ color: "#b0bdd2" }} />
//             </ListItemIcon>
//             <ListItemText
//               primary="Consult Efficiency"
//               classes={{
//                 root: classes.listItemTextRoot,
//                 primary: classes.listItemTextLabel
//               }} />
//           </ListItem>
//         </NavLink>
//       </List>
//     </Drawer>
//   );
// };

// export default withStyles(styles)(DashboardSideBar);
