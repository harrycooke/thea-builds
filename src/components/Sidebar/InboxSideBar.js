import React from 'react';
import {
  Card,
  List,
  Typography,
  ListItem,
  Divider,
  ListItemIcon,
  ListItemText
} from "@material-ui/core";
import { makeStyles } from '@material-ui/core/styles';
import QuestionAnswerIcon from '@material-ui/icons/QuestionAnswer';
import FolderIcon from '@material-ui/icons/Folder';
import { Link } from "react-router-dom";
import DraftsIcon from '@material-ui/icons/Drafts'
const drawerWidth = 200;

const useStyles = makeStyles(theme => ({
  root: {
    display: 'flex',
    maxWidth: drawerWidth,
    margin: 0,
    padding: 0,
    height: "100vh",
    paddingTop: theme.spacing(8),
  },
  drawer: {
    width: drawerWidth,
    flexShrink: 0, 
    // backgroundColor: "#06596b",
    backgroundColor: theme.palette.secondary.main

  },
  content: {
    flexGrow: 1,
    padding: 0,
  },
  toolbar: theme.mixins.toolbar,
  listItemTextRoot: {
    padding: 0
  },
  listItemTextLabel: {
    //color: "#7e8fa7",
    fontSize: 15,
    fontWeight: 600
  }
}));

export default function InboxSideBar(props) {
  const classes = useStyles();
  let repVal = ''

  const routeName = props.routeName.substr(props.routeName.length-1,1);
  if(routeName === '/'){
    repVal = props.routeName.substr(0,props.routeName.length-1);
  }else{
    repVal = props.routeName;
  }

  return (
    <Card className={classes.root}>
      <div className={classes.drawer}>
        <List>
          {/* <Divider variant="fullWidth" style={{ marginTop: 12 }}/> */}
          <Typography color="textSecondary" variant="subtitle1" style={{ marginLeft: 14, marginTop: 45, marginBottom: 10, fontWeight: 800 }}>
            FOLDERS
          </Typography>
          {/* <Divider variant="fullWidth" style={{ marginBottom: 10 }}/> */}
          
          <Link to="/inbox/cases">
            <ListItem button>
              <ListItemIcon>
                <QuestionAnswerIcon style={{ color: `${repVal === '' || repVal === '/inbox' || repVal === '/inbox/cases' ? '#fff' : '#87b9c4'}` }} />
              </ListItemIcon>
              <ListItemText style={{ color: `${repVal === '' || repVal === '/inbox' || repVal === '/inbox/cases' ? '#fff' : '#87b9c4'}` }}
                primary="Open Cases"
                classes={{
                  root: classes.listItemTextRoot,
                  primary: classes.listItemTextLabel
                }} />
            </ListItem>
          </Link>
          <Link to="/inbox/closed_cases">
            <ListItem button>
              <ListItemIcon>
                <FolderIcon style={{ color: `${repVal === '/inbox/closed_cases' ? '#fff' : '#87b9c4'}` }} />
              </ListItemIcon>
              <ListItemText style={{ color: `${repVal === '/inbox/closed_cases' ? '#fff' : '#87b9c4'}` }}
                primary="Closed Cases"
                classes={{
                  root: classes.listItemTextRoot,
                  primary: classes.listItemTextLabel
                }} />
            </ListItem>
          </Link>
          {props.draftExist &&
            <Link to="/inbox/drafts">
              <ListItem button>
                <ListItemIcon>
                  <DraftsIcon style={{ color: `${repVal === '/inbox/drafts' ? '#fff' : '#87b9c4'}` }} />
                </ListItemIcon>
                <ListItemText style={{ color: `${repVal === '/inbox/drafts' ? '#fff' : '#87b9c4'}` }}
                  primary="Drafts"
                  classes={{
                    root: classes.listItemTextRoot,
                    primary: classes.listItemTextLabel
                  }} />
              </ListItem>
            </Link>}
        </List>
      </div>
    </Card>
  );
}