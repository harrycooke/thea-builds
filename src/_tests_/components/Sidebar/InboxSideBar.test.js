import React from 'react';
import ReactDOM from'react-dom';

import TestRenderer from 'react-test-renderer'; // ES6
import sinon from 'sinon';
import {shallow, configure, mount, render} from 'enzyme';

import {InboxSideBar} from '../../../components/Sidebar/InboxSideBar';

describe('InboxSideBar', () => {
    // beforeAll(() => {
    //     this.commonProps = {};
    // });


    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(InboxSideBar, div);
    });

    it('renders React components, traversing a ReactElement tree and represents as JSON object', () => {
        const renderer = TestRenderer.create(InboxSideBar);
        renderer.toJSON();
      });
      
    describe('InboxSideBar', () => {
        it('should render correctly in "debug" mode', () => {
          const component = (InboxSideBar);
          expect(component).toMatchSnapshot();
        });
      });
});