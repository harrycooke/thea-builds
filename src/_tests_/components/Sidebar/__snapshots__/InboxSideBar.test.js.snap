// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InboxSideBar InboxSideBar should render correctly in "debug" mode 1`] = `undefined`;
